AnalysisTools/plugins/PATObjectValueEmbedder.cc

```diff
@@ -0,0 +1,154 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+//    PATObjectValueEmbedder                                                //
+//                                                                           //
+//    Takes a double and a collection of PAT objects and embeds the          //
+//    double as a userfloat in the objects                                   //
+//                                                                           //
+//    Nate Woods, U. Wisconsin                                               //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+
+
+// system includes
+#include <memory>
+#include <vector>
+#include <iostream>
+
+// CMS includes
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "DataFormats/PatCandidates/interface/Electron.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "DataFormats/PatCandidates/interface/Tau.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "DataFormats/PatCandidates/interface/Photon.h"
+#include "DataFormats/PatCandidates/interface/CompositeCandidate.h"
+#include "DataFormats/Common/interface/View.h"
+
+
+template<typename T, typename V>
+class PATObjectValueEmbedder : public edm::stream::EDProducer<>
+{
+
+public:
+  explicit PATObjectValueEmbedder(const edm::ParameterSet& iConfig);
+  virtual ~PATObjectValueEmbedder() {};
+
+private:
+  virtual void produce(edm::Event& iEvent, const edm::EventSetup& iSetup);
+  void embedValue(T& object, const int& value, const std::string& label) const;
+  void embedValue(T& object, const bool& value, const std::string& label) const;
+  void embedValue(T& object, const double& value, const std::string& label) const;
+  void embedValue(T& object, const float& value, const std::string& label) const;
+
+  const edm::EDGetTokenT<edm::View<T> > srcToken_;
+  const edm::EDGetTokenT<V> valToken_;
+  const std::string label_;
+};
+
+
+template<typename T, typename V>
+PATObjectValueEmbedder<T,V>::PATObjectValueEmbedder(const edm::ParameterSet& iConfig) :
+  srcToken_(consumes<edm::View<T> >(iConfig.getParameter<edm::InputTag>("src"))),
+  valToken_(consumes<V>(iConfig.getParameter<edm::InputTag>("valueSrc"))),
+  label_(iConfig.getParameter<std::string>("label"))
+{
+  produces<std::vector<T> >();
+}
+
+
+template<typename T, typename V>
+void PATObjectValueEmbedder<T,V>::produce(edm::Event& iEvent,
```

Comment:
As a future project, it would be cool to allow multiple values to be embedded at once, as you did in the ValueMap embedder. It's not a necessity as it was with ValueMaps, but it would be nice to skip the overhead of creating all the extra collections.